### DISCUSSION TITLE
Modify `read-only:` variant behavior

### DISCUSSION THREAD

**Discussion by neupauer** [tailwindlabs/tailwindcss#8097] [Ideas]
Hi, thing I learned today is that the [`:read-only`](https://developer.mozilla.org/en-US/docs/Web/CSS/:read-only) pseudo class targets all elements that is not editable by the user. That includes `div`s, `disabled` elements etc. 

![Screen Shot 2022-04-12 at 17 26 11](https://user-images.githubusercontent.com/6411116/162997877-0869b344-d810-4a2e-93c8-009af2c6f007.png)

My idea is to change the behavior of the existing `read-only:` variant to targets only elements with the `[readonly]` attribute (not `:read-only` pseudo class). I think such behavior will be more intuitive. Technically it is a breaking change, but I think it's worth considering.

Links:
https://play.tailwindcss.com/cbk8y7kCX4

**Comment by cvharris** [comment]
This bit me when I realized `<select>` is always read-only even without the attribute. While this follows the spec on the CSS selector, it's more a poor implementation in the whatwg and csswg specs more than anything
